code/modules/projectiles/guns/ballistic/shotgun.dm

```diff
@@ -118,6 +118,94 @@
 		return
 	rack()
 
+//The Riot Blaster
+
+/obj/item/gun/ballistic/shotgun/blaster
+	name = "riot blaster energy shotgun"
+	desc = "The Aussec riot blaster energy shotgun utilizes bluespace rifling to reduce friction and drag significantly, propelling bullets at much faster speeds than conventional shots. This requires power upkeep."
+	icon_state = "meltra" //To be replaced, don't stress this being misnamed.
+	inhand_icon_state = "meltra"
+	worn_icon_state = "meltra"
+	fire_delay = 8
+	mag_type = /obj/item/ammo_box/magazine/internal/shot/blaster
+	can_modify_ammo = TRUE
+	initial_caliber = CALIBER_SHOTGUN_ENERGY
+	alternative_caliber = CALIBER_SHOTGUN
+	alternative_ammo_misfires = TRUE
+	can_misfire = FALSE
+	misfire_probability = 0
+	misfire_percentage_increment = 25
+	projectile_damage_multiplier = 0.50
+
+	///Our cell for keeping our shotgun charged
+	var/obj/item/stock_parts/cell/cell
+	var/cell_type = /obj/item/stock_parts/cell
+
+	///Vars for tracking power expenditure
```

Comment:
this isnt how you should use autodocs, look at other files

Reply:
Damnit I've been deceived from like a whole year ago